completions.go

```diff
@@ -1,34 +1,88 @@
 package kingpin
 
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type completion struct {
```

Comment:
Yep, the public nature of the completion methods was a miss on my part.  I'll fix this and add documentation.